### PR TITLE
rem/aufile: fix aufile_get_length calculations

### PR DESCRIPTION
**Pull request by larsimmisch**
Also, `aufile_prm` should be const.

### CODE REVIEW

File: rem/aufile/aufile.c [hunk 1]
```diff
@@ -252,26 +252,26 @@ size_t aufile_get_size(struct aufile *af)
...
+	if (!af || !prm)
 		return 0;
 
 	switch (prm->fmt) {
 		case AUFMT_PCMA:
 		case AUFMT_PCMU:
-			return af->datasize * prm->channels * prm->srate
-				/ 1000;
+			return af->datasize / prm->channels * 1000
+				/ prm->srate;
```
Comment (cspiel1) [inline rem/aufile/aufile.c:264]: Maybe more readable and only one division:
```c
return af->datasize * 1000 / (prm->channels * prm->srate);
```

Comment (cspiel1) [reply rem/aufile/aufile.c:264]: Should we also take the WAV header size into account?
See function `aufile_set_position()`!

**Comment by sreimers** [comment]
A test case with a small wav file within `test/data` directory would be nice.

File: rem/aufile/aufile.c (hunk 1 above)
Comment (larsimmisch) [reply rem/aufile/aufile.c:264]: `af->datasize` is already the size of the data chunk without the header

File: test/aulength.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+
+#define DEBUG_MODULE "aulength"
+#define DEBUG_LEVEL 5
+#include <re_dbg.h>
+
+
+int test_aulength(void)
+{
+	struct aufile *af;
```
Comment (sreimers) [inline test/aulength.c:19]: initialize with NULL here. So mem_deref is fine if aufile_open fails.

Comment (larsimmisch) [reply test/aulength.c:19]: Good point

**Comment by sreimers** [comment]
Looks like microsoft has a broken repo currently:

```
E: Repository 'https://packages.microsoft.com/ubuntu/22.04/prod jammy InRelease' changed its 'Origin' value from 'microsoft-ubuntu-jammy-prod jammy' to 'Pulp 3'
E: Repository 'https://packages.microsoft.com/ubuntu/22.04/prod jammy InRelease' changed its 'Label' value from 'microsoft-ubuntu-jammy-prod jammy' to ''
Error: Process completed with exit code 100.
```

**Pull request merged**